homeassistant/components/evohome/__init__.py

```diff
@@ -388,27 +370,105 @@ def temperature_unit(self) -> str:
         """Return the temperature unit to use in the frontend UI."""
         return TEMP_CELSIUS
 
-    async def _call_client_api(self, api_function) -> None:
+    async def _call_client_api(self, api_function, refresh=True) -> Any:
         try:
-            await api_function
+            result = await api_function
         except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
-            _handle_exception(err)
+            if not _handle_exception(err):
+                return
 
-        self.hass.helpers.event.async_call_later(
-            2, self._evo_broker.update()
-        )  # call update() in 2 seconds
+        if refresh is True:
+            self.hass.helpers.event.async_call_later(1, self._evo_broker.update())
 
-    async def _update_schedule(self) -> None:
-        """Get the latest state data."""
-        if (
-            not self._schedule.get("DailySchedules")
-            or parse_datetime(self.setpoints["next"]["from"]) < utcnow()
-        ):
+        return result
+
+
+class EvoChild(EvoDevice):
+    """Base for any evohome child.
+
+    This includes (up to 12) Heating Zones and (optionally) a DHW controller.
+    """
+
+    def __init__(self, evo_broker, evo_device) -> None:
+        """Initialize the evohome Controller (hub)."""
+        super().__init__(evo_broker, evo_device)
+        self._schedule = self._setpoints = {}
```

Comment:
This makes it so that `_schedule` and `_setpoints` point at the exact same dictionary, and so any changes are shared too. Is that on purpose ?

Reply:
My understanding is not:
```
(venv) dbonnes@vm-builder:~/clients/somecomfort$ python
Python 3.6.8 (default, Aug 20 2019, 17:12:48)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = b = {}
>>> a = {'a': 1}
>>> a
{'a': 1}
>>> b
{}
>>> c = d = {'a': 1}
>>> c = {'a': 3}
>>> c
{'a': 3}
>>> d
{'a': 1}
```

Maybe was the case for older version of Python?

In any case, I will remove them.

Reply:
Uuuuh, what Python do you use 👀 

```
› python3
Python 3.7.4 (default, Jul  9 2019, 18:13:23)
[Clang 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> a = b = {}
>>> a[1] = True
>>> b
{1: True}
>>> a
{1: True}
>>>
```

Reply:
In my heart, I knew it was wrong! 

But it definitely worked for me, so I'd convinced myself that {} was immutable. :(

Reply:
In my heart, I knew it was wrong! 

But it definitely worked for me, so I'd convinced myself that {} was immutable. :(